app/javascript/mastodon/reducers/compose.js

```diff
@@ -122,6 +125,7 @@ function clearAll(state) {
     map.set('is_changing_upload', false);
     map.set('in_reply_to', null);
     map.set('privacy', state.get('default_privacy'));
+    map.set('federation', state.get('default_federation'));
```

Comment:
Consider making this an ENV var for admins to choose the default federation level. We also may want to make this a user level option the same way privacy level does.